crates/sui-indexer-alt-framework/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -255,3 +255,303 @@ pub(super) fn commit_watermark<H: Handler + 'static>(
         );
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{pipeline::Processor, MIGRATIONS};
+    use prometheus::Registry;
+    use std::time::Duration;
+    use sui_field_count::FieldCount;
+    use sui_pg_db::{self as db, temp::TempDb, Db, DbArgs};
+    use tokio::sync::mpsc;
+
+    struct TestHandler;
+
+    struct Entry;
+
+    impl FieldCount for Entry {
+        const FIELD_COUNT: usize = 1;
+    }
+
+    impl Processor for TestHandler {
+        const NAME: &'static str = "test";
+
+        type Value = Entry;
+
+        fn process(
+            &self,
+            _checkpoint: &Arc<sui_types::full_checkpoint_content::CheckpointData>,
+        ) -> anyhow::Result<Vec<Self::Value>> {
+            Ok(vec![])
+        }
+    }
+
+    #[async_trait::async_trait]
+    impl Handler for TestHandler {
+        async fn commit(
+            _values: &[Self::Value],
+            _conn: &mut db::Connection<'_>,
+        ) -> anyhow::Result<usize> {
+            Ok(0)
+        }
+    }
+
+    async fn test_db() -> (TempDb, Db) {
+        let temp_db = TempDb::new().unwrap();
+        let db_config = DbArgs::new(temp_db.database().url().clone());
+        let db = Db::new(db_config).await.unwrap();
+        db.run_migrations(MIGRATIONS).await.unwrap();
+        (temp_db, db)
+    }
+
+    // Create a new commit watermark with the given checkpoint.
+    // For the testing done here, only checkpoint matters so all other fields are arbitrary.
+    async fn new_commit_watermark(checkpoint: i64) -> CommitterWatermark<'static> {
+        CommitterWatermark {
+            pipeline: "test".into(),
+            epoch_hi_inclusive: 1,
+            checkpoint_hi_inclusive: checkpoint,
+            tx_hi: 10,
+            timestamp_ms_hi_inclusive: 100,
+        }
+    }
+
+    #[tokio::test]
+    async fn test_commit_watermark_basic() {
+        let (_temp_db, db) = test_db().await;
+        let registry = Registry::new_custom(Some("indexer_alt".to_string()), None).unwrap();
+        let metrics = Arc::new(IndexerMetrics::new(&registry));
```

Comment:
There is a `test_metrics` helper function in `metrics::test` which you could use here.